### PR TITLE
Update appveyor runner to Visual Studio 2022

### PR DESCRIPTION
**Pull request by zewpo**
This PR is an example how I got it working in the appveyor runner image VS 2022, if you want to upgrade to that.

Note: I moved stuff of the C:\ drive, into a .\DEPS\ folder, because I was having issues with conflicting pre-installed software in the runner image - already installed in those locations.   In hindsight, the main issue was with strawberry perl - and I might not have needed to move the all into DEPS.   Point is, it might be possible to revert some of the dependencies to pre-installed locations and make it run quicker. 

This is an alternative bigger version of CR #295 , and also Potentially Closes https://github.com/vim/vim-win32-installer/issues/294  ,

### CODE REVIEW

**Comment by zewpo** [comment]
This one is failing tests on appveyor.  strange thing is - this same code passes testing on github CI.  Not sure what the difference is yet.

**Comment by zewpo** [comment]
Update, I'm about to submit a PR to vim master, to resolve the appveyor test issue, and when/if that is accepted that should remove the roadblock that is preventing us from upgrading the builder image to VS 2022.

**Comment by zewpo** [comment]
Note - regarding the test failure.   I have already tested that it works by replicating this, and pulling from my vim branch.

It will continue to fail until https://github.com/vim/vim/pull/11793 is accepted.  Then it just needs a bump and it should work.

**Comment by chrisbra** [comment]
sure thanks!

**Comment by zewpo** [comment]
Update;   The PR blocking this is now; https://github.com/vim/vim/pull/11798

This PR no longer technically relies on https://github.com/vim/vim/pull/11793  because I've split that into two PRs.  Hopefully this makes it easier for @brammool  to manage any risks.

**Comment by zewpo** [comment]
This is passing the tests now that [9.0.1169](https://github.com/vim/vim-win32-installer/pull/296/commits/3c92e483c834c1c3131b199a4810bd698ce487b7) has been completed.

File: appveyor.bat [hunk 1]
```diff
@@ -112,13 +112,22 @@ echo TAG_NAME: %TAG_NAME%
...
 popd
 
 if not exist downloads mkdir downloads
+if not exist dependencies mkdir dependencies
+
+:: Update cygwin and add dependencies
+if not exist %CYGWIN_DIR% mkdir %CYGWIN_DIR%
+call :downloadfile %CYGWIN_URL% %CYGWIN_DIR%\setup-x86_64.exe
+%CYGWIN_DIR%\setup-x86_64.exe -qnNdO -P jq,make,gettext-devel
+:: Initialise the new bash profile, just in case it it a new install.
```
Comment (k-takata) [inline appveyor.bat:129]: > it it

typo?

File: appveyor.bat [hunk 2]
```diff
@@ -103,7 +103,7 @@ goto %target%_%ARCH%
 echo Unknown build target.
 exit 1
 
-
+
```
Comment (k-takata) [inline appveyor.bat:106]: Could you revert this change?

File: appveyor.yml [hunk 3]
```diff
@@ -65,7 +71,7 @@ deploy:
     # method: POST
     authorization:
       secure: eX3iWU3dQqDdg8UHR7Br6tqtvlFlhXihHcV0y/oR2YhSj6XZ2Pl/KVLiczUBCc39WWG/Aa5AXafWxaHQC/s40g==
-
+
```
Comment (k-takata) [inline appveyor.yml:74]: Could you revert this change?

File: appveyor.bat (hunk 1 above)
Comment (zewpo) [reply appveyor.bat]: yeah, sorry. fixed.

**Comment by zewpo** [comment]
@k-takata 
OK, I fixed each format and typo issue - using the github website interface.    Sorry that I didnt think about it queuing a new job for each little change - that might take over an hour to get through all that, irrelevant changes to the functionality.

**Comment by chrisbra** [comment]
so is this ready to be merged? Opinions?

**Comment by zewpo** [comment]
Yeah, I think so.  I fixed the formatting and typos that @k-takata  pointed out.

**Comment by chrisbra** [comment]
okay, I merged it in as ab0ebe1de8e9770ef441806c5e4ad10a49551243
